### ISSUE TITLE
[iOS / Forms] Strange transaction ID's from the API (or Plugin)?

### ISSUE THREAD

**Issue by Sebastian1989101** [jamesmontemagno/InAppBillingPlugin#252]
Using the `IInAppBillingVerifyPurchase` to verify the purchases on my servers, I receive some strange transaction ID's lately. They do look like GUID's and not like actual transaction ID's and generate an error afterwards. 

For this data on from the Apple API: 

```
{
  "receipt_type": "Production",
  "adam_id": 9998936999,
  "app_item_id": 9998936999,
  "bundle_id": "com.SoftwareNotion.MyApp",
  "application_version": "19273",
  "download_id": 99929637826999,
  "version_external_identifier": 832923544,
  "receipt_creation_date": "2019-10-17 02:24:49 Etc/GMT",
  "receipt_creation_date_ms": "1571279089000",
  "receipt_creation_date_pst": "2019-10-16 19:24:49 America/Los_Angeles",
  "request_date": "2019-10-17 02:24:57 Etc/GMT",
  "request_date_ms": "1571279097889",
  "request_date_pst": "2019-10-16 19:24:57 America/Los_Angeles",
  "original_purchase_date": "2017-04-12 04:00:48 Etc/GMT",
  "original_purchase_date_ms": "1491969648000",
  "original_purchase_date_pst": "2017-04-11 21:00:48 America/Los_Angeles",
  "original_application_version": "17094",
  "in_app": [
    {
      "quantity": "1",
      "product_id": "com.softwarenotion.myapp.allpages",
      "transaction_id": "99900394522999",
      "original_transaction_id": "99900394522999",
      "purchase_date": "2017-12-12 15:06:39 Etc/GMT",
      "purchase_date_ms": "1513091199000",
      "purchase_date_pst": "2017-12-12 07:06:39 America/Los_Angeles",
      "original_purchase_date": "2017-12-12 15:06:39 Etc/GMT",
      "original_purchase_date_ms": "1513091199000",
      "original_purchase_date_pst": "2017-12-12 07:06:39 America/Los_Angeles",
      "is_trial_period": "false"
    },
    {
      "quantity": "1",
      "product_id": "com.softwarenotion.myapp.removeads",
      "transaction_id": "99900394524999",
      "original_transaction_id": "99900394524999",
      "purchase_date": "2017-12-12 15:10:43 Etc/GMT",
      "purchase_date_ms": "1513091443000",
      "purchase_date_pst": "2017-12-12 07:10:43 America/Los_Angeles",
      "original_purchase_date": "2017-12-12 15:10:43 Etc/GMT",
      "original_purchase_date_ms": "1513091443000",
      "original_purchase_date_pst": "2017-12-12 07:10:43 America/Los_Angeles",
      "is_trial_period": "false"
    }
  ]
}
```

The app tries to request the verification with this transaction ID: `FFF21AF0-0335-4FE4-9EEC-FDF8FDC16999`. Why does the Plugin give me such a strange transaction ID? 

ps. I changed each ID first three and last three characters to protect the real data. So just ignore the "999" or "FFF" in the ID's those are actual different in the real production data. 

## Bug Information

Version Number of Plugin: 2.1.0.187-beta
Device Tested On: - 
Simulator Tested On: - 
Version of VS: Latest stable
Version of Xamarin: Latest stable
Versions of other things you are using: -

**Comment by jamesmontemagno** [comment]
Hmm very strange. Do you think that there is some change in iOS 13? Are you able to log out the device info and make a correlation?

We haven't changed anything and it still returns: Id = p.TransactionIdentifier, for the identifier.

https://github.com/jamesmontemagno/InAppBillingPlugin/blob/master/src/Plugin.InAppBilling.iOS/InAppBillingImplementation.cs#L502

**Comment by Sebastian1989101** [comment]
@jamesmontemagno This error don't occur on any of my devices which are running iOS 13 but I usually use TestFlight builds. Currently I don't track any user info (except a purchase verification fails but in this case only on server side). 

Edit: I just tested it with the device from my wife (iPhone 11 with iOS 13.1.2) and the purchase works without any failure so it requested with the expected transaction ID to my server. Also the failed requests are from all over the world so it is also not a local problem bound to a region. 

Based on the application revision in the response, it seems to only affect versions build after the iOS 13 release. Was there maybe some kind of change in Xamarin.iOS that could be the cause of this? Because I keep my build chain up2date with the latest stables whenever possible. The oldest revision I can find in my logs is 19273 (my revisions are two digits for the year and then three digits for the day in the year, so 19273 was built at 1st October 2019). The newest revision is 19288 (16th October 2019) so it seems like only october builds are affected by it.

**Comment by AntRemo** [comment]
Hi @Sebastian1989101 

Are you still experiencing these strange Transaction Ids?

**Comment by Sebastian1989101** [comment]
@AntRemo Yes it still happens but I still don't know what causes this. I added a workaround in my app so that my app will trigger a "Restore" directly after a Purchase. This way it seems to work in most cases.

**Comment by AntRemo** [comment]
@Sebastian1989101 
Interesting. Thanks. I'll post if I see the same occur.

**Comment by jamesmontemagno** [comment]
Closing due to age of issue and mass chagnes in library
